sdk/maps/Azure.Maps.Weather/src/MapsWeatherClient.cs

```diff
@@ -0,0 +1,1413 @@
+        /// <returns> The response returned from the service. </returns>
+        public virtual Response<QuarterDayForecastResult> GetQuarterDayForecast(GetQuarterDayForecastOptions options, CancellationToken cancellationToken = default)
+        {
+            Common.Argument.AssertNotNull(options.Coordinates, nameof(options.Coordinates));
+
+            using var scope = _clientDiagnostics.CreateScope("MapsWeatherClient.GetQuarterDayForecast");
+            scope.Start();
+            try
+            {
+                var coord = new[]
+                {
+                    Convert.ToDouble(options.Coordinates.Latitude, CultureInfo.InvariantCulture.NumberFormat),
+                    Convert.ToDouble(options.Coordinates.Longitude, CultureInfo.InvariantCulture.NumberFormat)
+                };
+                return restClient.GetQuarterDayForecast(JsonFormat.Json, coord, options.Unit, options.Duration, options.Language.ToString(), cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// [Protocol Method] Use to get current weather conditions.
+        /// <list type="bullet">
+        /// <item>
+        /// <description>
+        /// This <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md">protocol method</see> allows explicit creation of the request and processing of the response for advanced scenarios.
+        /// </description>
+        /// </item>
+        /// </list>
+        /// </summary>
+        /// <param name="options"> Additional options. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The response returned from the service. </returns>
+        public virtual async Task<Response<CurrentConditionsResult>> GetCurrentConditionsAsync(GetCurrentConditionsOptions options, CancellationToken cancellationToken = default)
+        {
+            Common.Argument.AssertNotNull(options.Coordinates, nameof(options.Coordinates));
+
+            using var scope = _clientDiagnostics.CreateScope("MapsWeatherClient.GetCurrentConditions");
+            scope.Start();
+            try
+            {
+                var coord = new[]
+                {
+                    Convert.ToDouble(options.Coordinates.Latitude, CultureInfo.InvariantCulture.NumberFormat),
+                    Convert.ToDouble(options.Coordinates.Longitude, CultureInfo.InvariantCulture.NumberFormat)
+                };
+                return await restClient.GetCurrentConditionsAsync(JsonFormat.Json, coord, options.Unit, options.Details == null ? "true" : (options.Details.Value ? "true" : "false"), options.Duration, options.Language.ToString(), cancellationToken).ConfigureAwait(false);
```

Comment:
Can we assign `options.Details` a default value to `true` in `GetCurrentConditionsOptions.cs`?
Something like: `public bool? Details { get; set; } = true;`?
And here the statement can be simplified as:
`options.Details?.Value ? "true" : "false"`